Content.Server/Radio/EntitySystems/TelecomSystem.cs

```diff
@@ -1,5 +1,35 @@
+using Content.Server.Power.Components;
+using Content.Server.Radio.Components;
+using Content.Shared.Radio.Components;
+
 namespace Content.Server.Radio.EntitySystems;
 
 public sealed class TelecomSystem : EntitySystem
 {
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RadioReceiveAttemptEvent>(OnRadioReceiveAttempt);
+    }
+
+    private void OnRadioReceiveAttempt(ref RadioReceiveAttemptEvent args)
+    {
+        if (args.Channel.longRange)
+            return;
+        if (HasComp<RadioMicrophoneComponent>(args.RadioSource) && HasComp<RadioSpeakerComponent>(args.RadioReceiver))
+            return;
+        var mapSource = Transform(args.RadioSource).MapID;
+        var mapReceiver = Transform(args.RadioReceiver).MapID;
+        if (mapSource == mapReceiver)
+        {
+            var map = mapSource;
```

Comment:
wut

Reply:
same check as for pinpointer for example, if maps are different then you can't send message

Reply:
You're copying mapSource into a local variable.

Reply:
well it doesn't matter, it's just to check if telecom server is on the same map as both sender and receiver

Reply:
It's unneeded so please remove it.